Code/Selenium_Advanced/SeleniumAdvancedEpamTests.cs

```diff
@@ -0,0 +1,145 @@
+                .Select(x=> x.Text);
+            foreach (var language in languagesArray)
+            {
+                Assert.IsTrue(foundLanduages.Contains(language), $"Language '{language}' was not found in the list.");
+            }
+        }
+        [Test]
+        public void NumberOfArticlesOnPageTest()
+        {
+            var searchIcon = _driver.FindElement(By.XPath("//span[contains(@class,'dark-iconheader-search__search-icon')]"));
+            searchIcon.Click();
+
+            /*ensuring that the elements from the Search pannel are present before proceeding further. Once the elements are found, 
+             * they are stored in the listOfSearchElements variable, which can be used in further code.  */
+            ReadOnlyCollection<IWebElement> listOfSearchElements = null;
+            _wait.Until(driver =>
+            {
+                listOfSearchElements = _driver.FindElements(By.XPath("//li[@class='frequent-searches__item']"));
+                return listOfSearchElements.Count > 0;
+            });
+
+            /*enerate a random index within the range of the list of search elements, checks if the element at that index is 
+             * displayed on the page, and clicks on it if it's displayed. This random selection and interaction with elements 
+             * continue until a displayed element is successfully clicked. */
+            Random random = new Random();
+            int randomIndex = random.Next(0, listOfSearchElements.Count);
+            _wait.Until(driver =>
+            {
+                if (listOfSearchElements[randomIndex].Displayed)
+                {
+                    listOfSearchElements[randomIndex].Click();
+                    return true;
+                }
+                return false;
+            });
+
+            var findButton = _driver.FindElement(By.XPath("//*[@class='bth-text-layer']"));
+            findButton.Click();
+
+            /*scrolling to ensure the last article is visible and checking for the presence of a "View More" link/button. 
+             * It continues waiting until the "View More" link/button is found.*/
+            _wait.Until(driver =>
+            {
+                var listOfArticles = _driver.FindElements(By.XPath("//article"));
+                action.ScrollToElement(listOfArticles[listOfArticles.Count - 1]).Build().Perform();
+                var searchViewMoreLink = _driver.FindElements(By.XPath("//a[@class='search-results__view-more button-text']"));
+
+                if (searchViewMoreLink.Count == 1)
+                {
+                    action.ScrollToElement(searchViewMoreLink[0]).Build().Perform();
+                }
+                return searchViewMoreLink.Count == 1;
+                });
+
+            var listOfArticles = _driver.FindElements(By.XPath("//article"));
+            Assert.That(listOfArticles.Count, Is.EqualTo(20));
```

Comment:
Do not hardcode values in assertions! + add assert message :)

Reply:
done